### ISSUE TITLE
aws_lambda: NODEJS_16 Runtime Not Available in Python CDK aws-cdk@2.41.0

### ISSUE THREAD

**Issue by saftinaz** [aws/aws-cdk#22021]
### Describe the bug

I have encountered a bug in which Python CDK version 2.41.0 is not able to create a Lambda function with NodeJS 16, the latest supported JavaScript runtime in the Lambda service.  

The runtime is however mentioned in the docs here:
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_lambda/Runtime.html#aws_cdk.aws_lambda.Runtime.NODEJS_16_X

Can we look into this?

### Expected Behavior

Be able to create a Lambda function in NodeJS 16 runtime with Python aws-cdk.

### Current Behavior

The following is my error:

```
File "....../cdk/backend_stack.py", line 83, in __init__
    runtime=lambda_.Runtime.NODEJS_16_X,
**AttributeError: type object 'Runtime' has no attribute 'NODEJS_16_X'. Did you mean: 'NODEJS_10_X'?**
```

### Reproduction Steps

Here is a reproduction of my code when I encountered this bug:

```
from aws_cdk import aws_lambda as lambda_
...
...
        self.function = lambda_.Function(
            self,
            id="Function",
            vpc=vpc,
            runtime=lambda_.Runtime.NODEJS_16_X,
            security_groups=[
                security_groups
            ],
            handler="index.handler",
            code=lambda_.Code.from_asset("..."),
        )
...
```

### CDK CLI Version

2.41.0 (build 6ad48a3)

### Node.js Version

Node v16.3.0

### OS

MacOS Monterey

### Language

Python

### Language Version

Python 3.10.4

**Comment by peterwoodworth** [comment]
This is working fine for me in python 2.41.0, could you try reinstalling your packages and verifying that all cdk packages installed are the same version?

**Comment by saftinaz** [comment]
Thank you for the response.

I updated cdk with `npm install aws-cdk -g --upgrade` on Node 16 and got the same error as above.
Then I updated to the latest Node 18.9.0 and ran `npm uninstall aws-cdk` and `npm install aws-cdk -g` again and still got the same error.

Here is the output of `cdk doctor` if it helps:
```% cdk doctor
ℹ️ CDK Version: 2.41.0 (build 6ad48a3)
ℹ️ AWS environment variables:
  - AWS_STS_REGIONAL_ENDPOINTS = regional
  - AWS_NODEJS_CONNECTION_REUSE_ENABLED = 1
  - AWS_SDK_LOAD_CONFIG = 1
ℹ️ No CDK environment variables```

What other information can I provide to help debug?
